### PR TITLE
Fix sort, remove user data on logout

### PR DESCRIPTION
**Pull request by kart-c**
## Minor Bug fixes and refactor

- Fix sort on products page - Previously products were sorted based on the initial price, now they are sorted on discounted price
- Remove the user's cart and wishlist on logout
- Fetch user's data back on successful login 
- Navigate the user to the last page on successful login
- Navigate the user to 2 pages back on successful signup
- Show the user's name in navbar upon login/signup

### CODE REVIEW

File: src/Utils/product-sort.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
 export const productSort = (state, products) => {
+	const discoutedPrice = (price, discount = 0) => price - (discount / 100) * price;
+
 	if (state.sortBy === 'lowtohigh') {
-		return [...products].sort((product1, product2) => product1.price - product2.price);
+		return [...products].sort(
+			(product1, product2) =>
+				discoutedPrice(product1.price, product1.discount) -
```
Comment (mohammed-sas) [inline src/Utils/product-sort.js]: typo "discounted"

**Review by mohammed-sas — APPROVED** [review]
Sort functionality is working fine
Navigating the user to the last page on successful login/signup is working fine

changes required
on successful signup , the username is displayed as "undefined" in the navbar.Fix this issue.

Other than suggested your PR is good to merge

**Review by AnkurChunekar — APPROVED** [review]
As suggested by @mohammed-sas, On signup username is displayed as "undefined", please look into this.
I think this is because you have used `name` as a property instead of `firstName` while executing the post request.
```
// src/Pages/Signup/Signup.jsx line: 30

const response = await axios.post('/api/auth/signup', {
    email: formData.email,
    password: formData.password,
    name: formData.name,
});
```
Rest everything looks good to merge.
- Sort Functionality is fixed and works fine.
- user's data persistence bug is fixed.
- User is correctly navigated back.
- code is very neat and clean.

File: src/Utils/product-sort.js (hunk 1 above)
Comment (kart-c) [reply src/Utils/product-sort.js]: Thank you @mohammed-sas . fixed it

**Comment by kart-c** [comment]
> As suggested by @mohammed-sas, On signup username is displayed as "undefined", please look into this. I think this is because you have used `name` as a property instead of `firstName` while executing the post request.
> 
> ```
> // src/Pages/Signup/Signup.jsx line: 30
> 
> const response = await axios.post('/api/auth/signup', {
>     email: formData.email,
>     password: formData.password,
>     name: formData.name,
> });
> ```
> 
> Rest everything looks good to merge.
> 
> * Sort Functionality is fixed and works fine.
> * user's data persistence bug is fixed.
> * User is correctly navigated back.
> * code is very neat and clean.

Thank you for the review, and the bug fix as well

**Pull request merged**